src/eventbridge/eventpattern/types.ts

```diff
@@ -0,0 +1,69 @@
+import { aws_events } from "aws-cdk-lib";
+
+/**
+ * CDK does not contain rich types for event bridge event patterns.
+ * https://github.com/aws/aws-cdk/issues/6184
+ */
+export interface ExistsPattern {
+  exists: boolean;
+}
+
+export const isExistsPattern = (x: any): x is ExistsPattern => {
+  return "exists" in x;
+};
+
+export interface PrefixPattern {
+  prefix: string;
+}
+
+export const isPrefixPattern = (x: any): x is PrefixPattern => {
+  return "prefix" in x;
+};
+
+/**
+ * Can only contain a single prefix pattern.
+ */
+export interface AnythingButPattern {
+  "anything-but": (string | number)[] | string | number | PrefixPattern | null;
+}
+
+export const isAnythingButPattern = (x: any): x is AnythingButPattern => {
+  return "anything-but" in x;
+};
+
+export interface NumberPattern {
+  number: [string, number, string, number] | [string, number];
+}
+
+export const isNumberPattern = (x: any): x is NumberPattern => {
+  return typeof x === "object" && "number" in x;
+};
+
+export const isMatchPattern = (x: any): x is MatchPattern => {
+  return x === null || typeof x === "string" || typeof x === "number";
+};
+
+export type MatchPattern = number | string | null | boolean;
+
+export interface SubPattern extends Record<string, Pattern> {}
+
+export type PatternList = (
+  | MatchPattern
+  | AnythingButPattern
+  | NumberPattern
+  | PrefixPattern
+  | ExistsPattern
+)[];
+
+export type Pattern = PatternList | undefined | SubPattern;
+
+/**
+ * As of CDK 2.14, Event Bridge doesn't support the content matchers.
```

Comment:
Doesn't our `peerDependency` prevent this from happening?

Reply:
Ohh, I meant that as of the current version of CDK, there is no support for content matchers in the types. There is no version that supports it now. I can remove the version to reduce confusion.